### PR TITLE
Generalizing JOSS

### PR DESCRIPTION
**Pull request by arfon**
👋 @parrish. I've moved this PR over to a branch on the parent repository so I can push updates to it too. Copying in your PR body from #353 here:

************

##### This looks like a larger PR than it really is since there's a fair bit of template reworking.  It's probably fairly digestable per-commit, though I'm happy to split this out into multiple PR's if it's easier.
----

Motivated by #329, this PR

- Loads the editor team from GitHub for the reviewer assignment and suggested reviewer on the submission.
- Generalizes the templates to use a [settings.yml](https://github.com/openjournals/joss/blob/ab305ef8c266dda60b33f3931c025d8ba9588a26/config/settings.yml) file to specify the easily extractable language and configuration
- Extracts as much customizable content from templates and template structure as possible.  These partials now live in [app/views/content](https://github.com/openjournals/joss/blob/ab305ef8c266dda60b33f3931c025d8ba9588a26/app/views/content)
- Adds a simple editors admin management UI at [http://staging.joss.theoj.org/editors](http://staging.joss.theoj.org/editors)
  - Can be bootstrapped via the [db/seeds](https://github.com/openjournals/joss/blob/ab305ef8c266dda60b33f3931c025d8ba9588a26/db/seeds.rb)
  - Currently, it's independent from the GitHub editor team, but that might be a cool feature.
  - It requires the admin flag on a user account, just ping me if you can't access it
- Adds a configurable list of [paper submission types](https://github.com/openjournals/joss/blob/ab305ef8c266dda60b33f3931c025d8ba9588a26/config/settings.yml#L20).
  - Each paper type will render a correspondingly named review checklist.
  - e.g. papers of type "awesome" would expect a checklist file at [app/views/content/github/_awesome_review_checklist.erb](https://github.com/openjournals/joss/blob/ab305ef8c266dda60b33f3931c025d8ba9588a26/app/views/content/github/_awesome_review_checklist.erb)

Along with openjournals/whedon-api#11, this should get us into a pretty good position to have multiple journals running. 🎉 

In-progress deploys are going to http://staging.joss.theoj.org

Feedback welcome!

### CODE REVIEW

File: app/views/shared/review_body.erb [hunk 1]
```diff
@@ -27,38 +28,8 @@ Please avoid lengthy details of difficulties in the review thread. Instead, plea
...
-
-### Software paper
-
-- [ ] **Authors:** Does the `paper.md` file include a list of authors with their affiliations?
-- [ ] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
-- [ ] **References:** Do all archival references that should have a DOI list one (e.g., papers, datasets, software)?
+<%- if paper.kind.present? %>
+  <%= render partial: "content/github/#{paper.kind}_review_checklist", locals: local_assigns %>
+<%- else %>
+  <%= render partial: "content/github/review_checklist", locals: local_assigns %>
```
Comment (arfon) [inline app/views/shared/review_body.erb]: Does `local_assigns` make the `paper` variable available to the partial at `content/github/review_checklist` @parrish?

File: db/seeds.rb [hunk 2]
```diff
@@ -1,7 +1,323 @@
...
+  description: <<-STR.strip_heredoc()
+    A lapsed academic with a passion for new models of scientific collaboration, he's used big telescopes
+    to study <a href="http://www.arfon.org/thesis" target="_blank">dust in space</a>, built
+    sequencing pipelines in <a href="http://www.sanger.ac.uk/" target="_blank">Cambridge</a> and
+    engaged millions of people in online citizen science by co-founding the
+    <a href="http://www.zooniverse.org" target="_blank">Zooniverse</a>.
+  STR
+)
+
+Editor.create(
```
Comment (arfon) [inline db/seeds.rb]: Shouldn't these all be `kind: topic`?

File: app/views/editors/_form.html.erb [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+  <div class="field">
+    <%= f.label :first_name %><br>
+    <%= f.text_field :first_name %>
+  </div>
+  <div class="field">
+    <%= f.label :last_name %><br>
+    <%= f.text_field :last_name %>
+  </div>
+  <div class="field">
+    <%= f.label :login %><br>
```
Comment (arfon) [inline app/views/editors/_form.html.erb]: I'm wondering if this should be a select based on `Repository.editors`

**Comment by arfon** [comment]
Oops. I just realized that you also merged your changes from #353 into #330.

**Comment by labarba** [comment]
Hi @arfon — when do you estimate that this PR might be merged in?
We're looking forward to kick starting JOSE!
//cc. @kyleniemeyer

File: app/views/shared/review_body.erb (hunk 1 above)
Comment (parrish) [reply app/views/shared/review_body.erb]: Yep, it's basically a passthrough for the locals that used to render shared/review_body

File: db/seeds.rb (hunk 2 above)
Comment (parrish) [reply db/seeds.rb]: Yeah, it's the column default, but there's no harm in being explicit here.

File: app/controllers/sessions_controller.rb [hunk 4]
```diff
@@ -1,6 +1,6 @@
 class SessionsController < ApplicationController
   def create
-    user = User.from_omniauth(env["omniauth.auth"])
+    user = User.from_omniauth(ENV["omniauth.auth"])
```
Comment (parrish) [inline app/controllers/sessions_controller.rb]: 🤦‍♂️

**Comment by arfon** [comment]
> Hi @arfon — when do you estimate that this PR might be merged in?
> We're looking forward to kick starting JOSE!

👋 @labarba @kyleniemeyer - we're getting _really_ close. I've been testing all of these changes over the last couple of days and things are looking pretty solid. I hope to finish up this testing this week.

**Comment by labarba** [comment]
Are we there yet?

File: db/seeds.rb [hunk 5]
```diff
@@ -1,7 +1,341 @@
...
+  kind: "topic",
+  first_name: "George",
+  last_name: "Githinji",
+  login: "biorelated",
+  email: "",
+  avatar_url: "http://joss.theoj.org/assets/editors/george-d24f98577bcc40a745b81f7ee5c4b6a3e5f150b563f940cb792ba92f88ed6483.jpg",
```
Comment (arfon) [inline db/seeds.rb]: @parrish - what should we do about URLs like this? How does encoding the asset pipeline stuff (i.e. `george-d24f98577bcc40a745b81f7ee5c4b6a3e5f150b563f940cb792ba92f88ed6483.png` work in the long run?

Comment (arfon) [reply db/seeds.rb]: I guess perhaps it doesn't matter as long as the file doesn't change? http://guides.rubyonrails.org/asset_pipeline.html#what-is-fingerprinting-and-why-should-i-care-questionmark

**Comment by arfon** [comment]
> Are we there yet?

I think so yes. I'm planning on migrating the JOSS site over to this updated code later today and once that's done I think we're ready to start rolling out the new JOSE codebase.

**Pull request merged**